### ISSUE TITLE
Is adding an "ongoing" property to catch up on the daily count a good idea?

### ISSUE THREAD

**Issue by HereIsZephyrus** [rschristian/github-contribution-calendar-api#6]
Hi, I'm very excited about this tool. It is convenient and helpful. Thank you so much.
I'm trying to draw a GitHub contribution heatmap in my blog site and I want to keep this renewed daily. And I found this API doesn't fetch the newest data. I viewed the JSON file structure and recognized that it is organized on a weekly basis as the data structure offered by GitHub API(https://github.com/users/{username}/contributions). So it doesn't contain the contribution data of the ongoing week that exists though.
The JSON file has two properties(total and contributions) as same as the GitHub API. Is adding an "ongoing" property to catch up on the daily count a good idea? This property will not destroy the current data structure and provide an alter for those who want to renew their contribution count daily.

**Comment by rschristian** [comment]
Glad you've found it useful.

> And I found this API doesn't fetch the newest data.

Yes, values are cached for 24 hours. If you want to blow through the cache you'll need to do that in your browser.

> So it doesn't contain the contribution data of the ongoing week that exists though.

Looks like a bug, it's not intentional.

**Comment by HereIsZephyrus** [comment]
Thanks for your reply.
> Looks like a bug, it's not intentional.

Haha, I thought before that it was to make each week's data equal in length to prevent unexpected behavior.

The structure of raw data is organized by the name of the day of the week rather not the time. So the data of the ongoing week will be cut off here I guess.
```
Object.values(data)
        .sort((a, b) => (a.date < b.date ? -1 : 1))
        .slice(0, limit * 7)
        .forEach((day, idx) => {
            weekIndex = Math.floor(idx / 7);
            if (!contributions[weekIndex]) contributions.push([]);
            contributions[weekIndex].push(day);
        });
```
I'm new to programming. Thanks for your help.

**Comment by rschristian** [comment]
Indeed, `limit` defaults to `-1` and so that slice call turns into `.slice(0, -7)`, hence the cut off. Oops

PRs welcome, if you wanted to fix it. Else I can take a look at some point.

**Comment by HereIsZephyrus** [comment]
That's great. Maybe I can have a try.
And just appending an array that is shorter then seven in the contribution property is recommended?

**Comment by rschristian** [comment]
Fixed in #7
